src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -153,7 +153,7 @@ internal static void ValidateImpersonationLevel(TokenImpersonationLevel imperson
 
         internal static Win32Exception CreateExceptionFromError(SecurityStatusPal statusCode)
         {
-            return new Win32Exception(NTE_FAIL, statusCode.ToString());
+            return new Win32Exception(NTE_FAIL, statusCode.Exception.Message);
```

Comment:
In the error scenario ,, where it will be getting called, we are setting the exception explicitly. Though it is good to make a null check to be safe.. I will do that,.